mqtt/mqtt-broker/src/broker.rs

```diff
@@ -991,15 +998,26 @@ where
         self
     }
 
+    pub fn with_config(mut self, config: BrokerConfig) -> Self {
+        self.config = Some(config);
+        self
+    }
+
     pub fn build(self) -> Broker<Z> {
+        let config = match self.config {
```

Comment:
lets put BrokerConfig::default() in the constructor instead?

Reply:
I think that this is not a big deal and would prefer to keep `default()` as a constructor for BrokerBuilder.

Reply:
I guess I didn't get what you meant the first time. Fixed it now.